bitcoin/src/blockdata/transaction.rs

```diff
@@ -53,21 +53,29 @@ impl_hashencode!(Txid);
 impl_hashencode!(Wtxid);
 
 impl Txid {
-    /// The "all zeros" TXID.
+    /// The `Txid` used in a coinbase prevout.
     ///
-    /// This is used as the "txid" of the dummy input of a coinbase transaction. It is
-    /// not a real TXID and should not be used in other contexts.
-    pub fn all_zeros() -> Self { Self::from_byte_array([0; 32]) }
+    /// This is used as the "txid" of the dummy input of a coinbase transaction. This is not a real
+    /// TXID and should not be used in any other contexts. See [`OutPoint::COINBASE_PREVOUT`].
+    pub const COINBASE_PREVOUT: Self = Self::from_byte_array([0; 32]);
+
+    /// The "all zeros" TXID.
+    #[deprecated(since = "TBD", note = "use Txid::COINBASE_PREVOUT instead")]
+    pub fn all_zeros() -> Self { Self::COINBASE_PREVOUT }
 }
 
 impl Wtxid {
-    /// The "all zeros" wTXID.
+    /// The `Wtxid` of a coinbase transaction.
     ///
     /// This is used as the wTXID for the coinbase transaction when constructing blocks,
     /// since the coinbase transaction contains a commitment to all transactions' wTXIDs
     /// but naturally cannot commit to its own. It is not a real wTXID and should not be
```

Comment:
I think the claim about it not being real confused me previously. It is real, it just must not be used outside of computing the commitment.

Reply:
Yeah, agreed. I wasn't thinking clearly when I wrote this. The wtxid is "the transaction ID used in the witness commitment tree" ... so if that ID is all zeros, that's what it is.

Reply:
How's this:

```rust
    /// This is used as the wTXID for the coinbase transaction when constructing blocks (in the
    /// witness commitment tree) since the coinbase transaction contains a commitment to all
    /// transactions' wTXIDs but naturally cannot commit to its own.
```

Reply:
I"ll do this as a follow up if its liked, we have one ack already and this PR is holding up #3288.

Reply:
That text looks great. Agreed, fine to do as a followup.